tools/static-assets/server/boot.js

```diff
@@ -193,47 +196,41 @@ var loadServerBundles = Profile("Load server bundles", function () {
        */
       require: Profile(function getBucketName(name) {
         return "Npm.require(" + JSON.stringify(name) + ")";
-      }, function (name) {
-        if (nonLocalNodeModulesPaths.length === 0) {
-          return require(name);
-        }
-
-        var fullPath;
-
-        nonLocalNodeModulesPaths.some(function (nodeModuleBase) {
-          var packageBase = files.convertToOSPath(files.pathResolve(
-            nodeModuleBase,
-            name.split("/", 1)[0]
-          ));
-
-          if (statOrNull(packageBase)) {
-            return fullPath = files.convertToOSPath(
-              files.pathResolve(nodeModuleBase, name)
-            );
+      }, function (name, error) {
+        if (nonLocalNodeModulesPaths.length > 0) {
+          var fullPath;
+
+          nonLocalNodeModulesPaths.some(function (nodeModuleBase) {
+            var packageBase = files.convertToOSPath(files.pathResolve(
+              nodeModuleBase,
+              name.split("/", 1)[0]
+            ));
+
+            if (statOrNull(packageBase)) {
+              return fullPath = files.convertToOSPath(
+                files.pathResolve(nodeModuleBase, name)
+              );
+            }
+          });
+
+          if (fullPath) {
+            return require(fullPath);
           }
-        });
+        }
 
-        if (fullPath) {
-          return require(fullPath);
+        var resolved = require.resolve(name);
+        if (resolved === name && ! path.isAbsolute(resolved)) {
+          // If require.resolve(id) === id and id is not an absolute
+          // identifier, it must be a built-in module like fs or http.
+          return require(resolved);
         }
 
-        try {
-          return require(name);
```

Comment:
Not calling `require(name)` unconditionally here is the key to keeping lookups from escaping the application.